pkg/controller/controller_utils.go

```diff
@@ -706,9 +707,8 @@ func (s ByLogging) Less(i, j int) bool {
 		return len(s[i].Spec.NodeName) > 0
 	}
 	// 2. PodRunning < PodUnknown < PodPending
-	m := map[v1.PodPhase]int{v1.PodRunning: 0, v1.PodUnknown: 1, v1.PodPending: 2}
-	if m[s[i].Status.Phase] != m[s[j].Status.Phase] {
-		return m[s[i].Status.Phase] < m[s[j].Status.Phase]
+	if podPhaseToOrdinal[s[i].Status.Phase] != podPhaseToOrdinal[s[j].Status.Phase] {
```

Comment:
One last thing:  Is it necessary to look up the value in the map for this comparison?  We need it for the `>` below (or `<` in the other `Less` methods), but I don't understand why this comparison isn't simply `s[i].Status.Phase != s[j].Status.Phase`.